public/src/map.js

```diff
@@ -35,7 +52,36 @@ export async function initMap(apiKey) {
   });
 
   // Add the geocoder to the map
-  map.addControl(geocoder);
+  // map.addControl(areaGeocoder);
+  map.addControl(constituencyGeocoder);
+
+  function dummy() {
+    console.log('dummy');
+    return [];
+  }
+  
+  function localSearch(query) {
+    const matchingFeatures = [];
+    return fetch('https://opendata.arcgis.com/datasets/937997590f724a398ccc0100dbd9feee_0.geojson')
```

Comment:
This seems super inefficient (fetching the entire GeoJSON file with every search) and as it's being fetched from a 3rd party service is a bit of an abuse of their resources

Smart browsers probably cache the data anyway and it isn't an issue, but it's not good to rely on that

Reply:
Not really sure how else to do it. Actually followed this tutorial [https://docs.mapbox.com/mapbox-gl-js/example/forward-geocode-custom-data/](url) and somehow still did not get things to work. I used a `const data = require('<link>')`.

Was googling for answers and somehow stumbled upon this..

Reply:
You managed to research and get it working which is the important thing, but I might see if the link there makes any sense to me later on as I wouldn't feel comfortable merging with the request like this

Reply:
Also, you can't use `require` in client-side code, it is a Node.js interface

Reply:
yep 👍 understood